Pw patch masses

Iximiel opened the pull request:
<!--
  Feel free to delete not relevant sections below
-->

##### Description

I Think I solved the problem with quantum espresso's pw don't communicating the correct masses of the atoms. I need the CI to test the patch

##### Target release

<!-- please tell us where you would like your code to appear (e.g. v2.4): -->
I may need to rebase these commit on 2.8 or older version, since the code change is very small

##### Type of contribution

<!--
  Please select the type of your contribution among these:
  (Change [ ] to [X] to tick an option)
-->
- [X] changes to code or doc authored by PLUMED developers, or additions of code in the core or within the default modules

##### Copyright

<!--
  In case you picked one of the first two choices
  MAKE SURE TO TICK ALSO THE FOLLOWING BOX
-->

- [X] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

<!--
  In case you picked the third choice (new module authored by you)
  MAKE SURE TO TICK ALSO THE FOLLOWING BOX
-->

##### Tests

<!--
  Make sure these boxes are checked. For Travis-CI tests, you can wait for them
  to be completed monitoring this page after your pull request has been submitted:
  http://travis-ci.org/plumed/plumed2/pull_requests
-->

- [ ] I added a new regtest or modified an existing regtest to validate my changes.
- [ ] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

<!--
  After your branch has been merged to the desired branch and then to plumed2/master, and after the
  plumed official manual has been updated, please check out the coverage scan at
  http://www.plumed.org/coverage-master
  In case your new features are not well covered, please try to add more complete regtests.
-->

GiovanniBussi commented:
Great! Yes I think it can be backported. Can you test it [here](https://github.com/plumed/test-plumed-qe)?

I guess you should be able to do it by:
- making a fork
- in your fork, modify [this line](https://github.com/plumed/test-plumed-qe/blob/621be366fcbb4361aaf228c104a3a6fe71ea09e2/.github/workflows/ci.yml#L77) so as to install the modified plumed patch
- modify the test, which currently has only two Si atoms, or better add a new test

Iximiel commented:
The [PR](https://github.com/plumed/test-plumed-qe/pull/4#issue-1796406567) for the test is here,
I also discovered that I have updated the patch for qe 6.2 and not the one for qe 7.0.
Moreover I corrected a few Fortran errors that I wrote.
